### ISSUE TITLE
Re-evaluate `fast` functionality

### ISSUE THREAD

**Issue by nycdotnet** [TypeStrong/grunt-ts#289]
The current `fast` and `watch` functionality may possibly be improved by considering integration with `--isolatedModules` and the native watch functionality.  Perhaps not.  Any thoughts are welcome.

//cc: @basarat

**Comment by jbrantly** [comment]
As I understand it grunt-ts actually executes `tsc` directly. I could be mistaken but I'm not sure that `--isolatedModules` actually provides any speed increase unless you're using the `transpileModule` TS API. I'm purely basing this off the comments [here](https://github.com/Microsoft/TypeScript/issues/2499) and [here](https://github.com/Microsoft/TypeScript/pull/3208). Namely:

> the idea is you would use this with in a normal compilation, with multiple files, to ensure you can do the compile them in isolation later, hence the "s".

**Comment by basarat** [comment]
Yeah, `isolatedModules` is mostly for error reporting about someone using things that not safe for transpilation. For `fast` we only give TypeScript one file ... so will not have any impact
